validator/client/src/main/java/tech/pegasys/teku/validator/client/SyncCommitteeScheduler.java

```diff
@@ -150,7 +150,15 @@ public void onHeadUpdate(
       final UInt64 slot,
       final Bytes32 previousDutyDependentRoot,
       final Bytes32 currentDutyDependentRoot,
-      final Bytes32 headBlockRoot) {}
+      final Bytes32 headBlockRoot) {
+
+    // recalculation of sync committee duties at the slot prior to the altair fork activation slot
+    if (spec.getSyncCommitteeUtil(slot).isEmpty()
+        && spec.getSyncCommitteeUtil(slot.plus(1)).isPresent()) {
```

Comment:
This isn't quite correct.  We should recalculate if we get a new head block anytime before the Altair fork is activated.  The epoch transition into Altair calculates the first two sync committees using the current randao value. So any new block received prior to that transition will change the randao value and thus change the sync committees (so we should recalculate duties).

This code as is will handle the case of the last block being received after we calculated duties but not the case where the second last block is received and the last slot remains empty.  I think we just want:
```suggestion
    if (spec.getSyncCommitteeUtil(slot).isEmpty()) {
```